nf_core/create.py

```diff
@@ -525,19 +527,22 @@ def git_init_pipeline(self):
         """
         # Check that the default branch is not dev
         try:
-            default_branch = git.config.GitConfigParser().get_value("init", "defaultBranch")
+            default_branch = self.default_branch or git.config.GitConfigParser().get_value("init", "defaultBranch")
         except configparser.Error:
             default_branch = None
```

Comment:
```suggestion
            default_branch = self.default_branch
```
`GitConfigParser` will fail if `defaultBranch` is not defined in the git config. If `self.default_branch` is specified we probably want to fall back on this value.

Reply:
If `self.default_branch` is truthy, the config parser will not run. See:

```
>>> True or print("falsy detected before or")
True
>>> False or print("falsy detected before or")
falsy detected before or
```

Reply:
This means if the default branch is set the config parser wont run and hence also not produce any exceptions.

Reply:
found an even nicer solution.